packages/eslint-plugin-pf-codemods/src/rules/v6/buttonMoveIconsIconProp/button-moveIcons-icon-prop.ts

```diff
@@ -87,6 +87,18 @@ module.exports = {
                         fixes.push(fixer.replaceText(child, ""))
                     );
                   }
+
+                  if (node.closingElement) {
+                    const closingSymbol = context
+                      .getSourceCode()
+                      .getLastToken(node.openingElement)!;
+
+                    fixes.push(
+                      fixer.replaceText(closingSymbol, " />"),
+                      fixer.remove(node.closingElement)
+                    );
+                  }
```

Comment:
Think it would be worth helperizing this? Seems like something that would be useful in a lot of situations.

Though if we do helperize it we should probably add a check to only make an element self closing if it doesn't have any children.

Reply:
I added the helper.

About checking that element has no children - this was problematic, because for example in this button rule, we remove the children in a fix and at the same time this fix for removing closing element is applied (so the node still has children at the time the condition is checked). We would have to run the self-closing fix additionaly in a cleanup rule.

So I made the helper customizable where it is not allowed to remove children by default, but it can be overwritten.